面談日程を承認or拒否できるようにする

mashvalue1 opened the pull request:
やりたかったこと
---

- 面談の承認
- 承認した面談の日時が過去の場合は編集画面へ遷移
- 面談が正常に承認された場合は他の面談は却下状態に更新

...

やったこと
----

- 面談承認のためのactionを定義
- 面談日時を選択した場合の確認メッセージの定義
- 面談日時更新処理の実装
- 面談日時更新の成否に応じた画面遷移
- 設定された面談日時をview側に反映


動作確認方法
---

- https://e-navigator-mashvalue1.herokuapp.com/ にアクセスしサインイン
- interview listからいずれかのユーザーの面談日時の一覧ページに移動
- いずれかの日時を選択すると確認メッセージが表示されるかを確認
- 選択した日時が未来の場合、user一覧ページにリダイレクトされ、該当ユーザーの面談日時が表示されていることを確認
- 選択した日時が過去の場合、エラーメッセージが表示され、日時編集画面へ移動することを確認


その他
---

何か分からない点や、ここ頑張りましたなどあればこちらに書いてください。

Commit:
update other_interviews-index

現在設定されている面談の有無で表示内容を分岐するように設定
helperにメソッドを定義しviewから呼び出し

Commit:
add setup action

面談を設定するアクションをルーティングに追加

Commit:
def date_format method

日時表示を返すメソッドをヘルパーに定義
日時を表示する部分で呼び出し

Commit:
add confirm message

日時を選択すると確認メッセージを表示するように設定

Commit:
add setup action

選択された日時の面談のステータスをacceptedに更新し、他の面談はrejectedに更新
選択した面談が過去の日付の場合はedit画面を表示

Commit:
update users index

ユーザー一覧で承認されている面談の日時を表示

Commit:
fix date_format method

引数が存在しない場合に対応できるように修正

Commit:
fix interviews index

ユーザー名を記入していない場合に表示を分岐

sukechannnn commented on app/controllers/interviews_controller.rb:
```diff
@@ -38,6 +38,17 @@ def destroy
+    if @interview.update(status: 1)
+      others.update_all(status: 2)
+      flash[:success] = "Interview has been set."
+      redirect_to root_url
```
ここ、アップデートした後は元いた画面`interviews`に戻る方が良いと思います！

あと余談ですが、ここで`root_url`の代わりに`root_path`を指定しても同じ動作をしますが、意味が微妙に違います。
`hoge_url`は絶対パスで`hoge_path`は相対パスです。redidectする場合は`hoge_path`の方をよく使います。使い分けられるとベストですね👍

mashvalue1 replied:
リダイレクト先を修正しました。
リダイレクトのurlとpathについては、railsチュートリアルがurlで記述していたのでurlを選んでいました。
どこかのタイミングで余裕があればpathに修正します。

sukechannnn commented on app/helpers/interviews_helper.rb:
```diff
@@ -1,2 +1,14 @@
+
+  def date_format(interview)
+    return interview.interview_date.strftime("%Y年%m月%d日 %H時%M分") if interview
+  end
```
こういった処理をきちんとhelperに定義してるの良いですね👏

sukechannnn commented on app/views/users/index.html.erb:
```diff
@@ -20,7 +20,7 @@
       <td><%= user.sex %></td>
       <td><%= user.school %></td>
-      <td>ここに面談日程</td>
+      <td><%= date_format(get_current_interview(user)) %></td>
```
TODOをしっかり回収してるの良いです👍

sukechannnn commented:
https://github.com/mashvalue1/e-navigator/pull/5#pullrequestreview-97071267 だけ修正お願いします！

Commit:
update redirect on setup-action, fix arg

面接設定後のリダイレクトを各ユーザーのinterviews#indexに設定
user_idをparamsから取得するように修正し、before_actionでの@user定義を除外

sukechannnn commented:
差分確認しました！
良いですね👍
LGTM

[![LGTM](https://www.lgtm.in/p/IuWAh93oq)](https://www.lgtm.in/i/IuWAh93oq)
